addons/account/models/company.py

```diff
@@ -278,6 +280,73 @@ def _validate_fiscalyear_lock(self, values):
+                if country_code in SYSCOHADA_LIST:
+                    #countries using OHADA Chart of Accounts
+                    module_list.append('l10n_syscohada')
+                elif country_code == 'GB':
+                    module_list.append('l10n_uk')
+                elif country_code == 'DE':
+                    module_list.append('l10n_de_skr03')
+                    module_list.append('l10n_de_skr04')
+                elif country_code == 'CN':
+                    module_list.append('l10n_cn_small_business')
+                    module_list.append('l10n_cn_standard')
+                else:
+                    if company.env['ir.module.module'].search([('name', '=', 'l10n_' + country_code.lower())]):
+                        module_list.append('l10n_' + country_code.lower())
+                    else:
+                        module_list.append('l10n_generic_coa')
+                if country_code == 'US':
+                    module_list.append('account_plaid')
+                    module_list.append('l10n_us_check_printing')
+                if country_code == 'CA':
+                    module_list.append('l10n_ca_check_printing')
+                if country_code in ['US', 'AU', 'NZ', 'CA', 'CO', 'EC', 'ES', 'FR', 'IN', 'MX', 'UK']:
+                    module_list.append('account_yodlee')
+                if country_code in SYSCOHADA_LIST + [
+                    'AT', 'BE', 'CA', 'CO', 'DE', 'EC', 'ES', 'ET', 'FR', 'GR', 'IT', 'LU', 'MX', 'NL', 'NO',
+                    'PL', 'PT', 'RO', 'SI', 'TR', 'UK', 'VE', 'VN'
+                    ]:
+                    module_list.append('base_vat')
+                if country_code == 'MX':
+                    module_list.append('l10n_mx_edi')
+
+                # European countries will be using SEPA
+                europe = company.env.ref('base.europe', raise_if_not_found=False)
+                if europe:
+                    europe_country_codes = [x.code for x in europe.country_ids]
+                    if country_code in europe_country_codes:
+                        module_list.append('account_sepa')
+                        module_list.append('account_bank_statement_import_camt')
+                module_ids = company.env['ir.module.module'].search([('name', 'in', module_list), ('state', '=', 'uninstalled')])
+                #module is being installed.
+                if(from_init):
+                    module_ids.sudo().button_install()
+                else:
+                    #a new company is being created.
+                    #localization for the given country isn't installed.
+                    if module_ids:
+                        module_ids.with_context({'allowed_company_ids': [company.id]}).button_immediate_install()
+                    #localizaton for the given country is already installed.
+                    else:
+                        #filter out the list of templates that belong to newly created company's country and load the first one.
```

Comment:
I didn't get you. This code is there to load appropriate templates for the new company. If we remove this, templates won't be loaded for new company even when they are installed.
kindly let me know If I misunderstood something. Thanks.

Reply:
@hdh-odoo In most localizations, there is a script that loads it automatically for the current company.  (try_loading...)

Reply:
@jva-odoo The logic should be different: 1 check if the module has been installed or not: if not, install / if already there: create a method to install the chart of accounts that localizations can inherit.  (with more or less in it what is below)

Reply:
@jco-odoo here 1 we check module is installed or not using "module_ids": after if not installed then we install using button_immediate_install else installed then we call try_loading_for_current_company.
Logic is the same as you say....so I didn't get where this is wrong or need improvement...!!?

Reply:
from_init or not, it should check if the module is installed or not.  And actually installing the CoA could be the same thing.  So, why do we need from_init in the first place?

Reply:
@jco-odoo from company create method if I call button_install then it's not installing a module so I need to call button_immediate_install instead of button_install.

Reply:
The button_immediate_install is needed (put it in a comment) for blocking concurrent access with the crons that might be running.